extensions/amp-list/0.1/amp-list.js

```diff
@@ -386,16 +390,76 @@ export class AmpList extends AMP.BaseElement {
       const event = createCustomEvent(this.win,
           AmpEvents.DOM_UPDATE, /* detail */ null, {bubbles: true});
       this.container_.dispatchEvent(event);
-
       // Change height if needed.
       this.measureElement(() => {
         const scrollHeight = this.container_./*OK*/scrollHeight;
         const height = this.element./*OK*/offsetHeight;
         if (scrollHeight > height) {
-          this.attemptChangeHeight(scrollHeight).catch(() => {});
+          if (resizeExperimentOn) {
+            // This needs to be gated behind the experiment 'amp-list-resize'
+            const layout = this.element.getAttribute('layout');
+            if (layout == Layout.NODISPLAY || layout == Layout.FILL) {
+              return;
+            } else if (layout == Layout.FLEX_ITEM) {
+              // TODO (cathyxz): flex item does not play nice
+              // with layout container
+              this.attemptChangeHeight(scrollHeight).catch(() => {});
+            } else if (layout !== Layout.CONTAINER) {
+              this.changeToLayoutContainer_();
+            }
+          } else {
+            this.attemptChangeHeight(scrollHeight).catch(() => {});
+          }
         }
       });
-    }, this.container_);
+    });
+  }
+
+  /**
+   * Converts the amp-list to de facto layout container.
+   * @private
+   */
+  changeToLayoutContainer_() {
+    const layout = this.element.getAttribute('layout');
+    if (layout == Layout.RESPONSIVE) {
+      const sizer = childElementByTag(this.element, 'i-amphtml-sizer');
```

Comment:
nit: this seems fine to do regardless of layout, similar to how you remove `'i-amphtml-layout-size-defined'` class regardless of layout after the `if else` block

Reply:
Done.